src/mxarray.jl

```diff
@@ -670,24 +672,26 @@ function jvariable(mx::MxArray)
             jsparse(mx)
         end
     elseif is_char(mx) && (nrows(mx) == 1 || is_empty(mx))
-        jstring(mx)
+        String(mx)
     elseif is_cell(mx)
         ndims(mx) == 2 ? (ncols(mx) == 1 ? jvector(mx) : jmatrix(mx)) :
         jarray(mx)
     elseif is_struct(mx) && nelems(mx) == 1
-        jdict(mx)    
+        jdict(mx)
+    elseif classid(mx) == mxFUNCTION_CLASS && nelems(mx) == 1
+        Function(mx)
     else
-        throw(ArgumentError("Unsupported kind of variable."))
+        throw(ArgumentError("No known conversion to a Julia value."))
     end
 end
 
 # deep conversion from MATLAB variable to Julia array
 
-jvariable(mx::MxArray, ty::Type{Array})  = jarray(mx)
-jvariable(mx::MxArray, ty::Type{Vector}) = jvector(mx)
-jvariable(mx::MxArray, ty::Type{Matrix}) = jmatrix(mx)
-jvariable(mx::MxArray, ty::Type{Number}) = jscalar(mx)::Number
-jvariable(mx::MxArray, ty::Type{String}) = jstring(mx)::String
-jvariable(mx::MxArray, ty::Type{Dict}) = jdict(mx)
-jvariable(mx::MxArray, ty::Type{SparseMatrixCSC}) = jsparse(mx)
-
+convert(::Type{Array}, mx::MxArray)  = jarray(mx)
+convert(::Type{Vector}, mx::MxArray) = jvector(mx)
+convert(::Type{Matrix}, mx::MxArray) = jmatrix(mx)
+convert(::Type{Number}, mx::MxArray) = jscalar(mx)::Number
```

Comment:
Did you only changed some of the signatures or is the RHS here missing from the change?